src/test/java/bio/terra/service/tags/TagUtilsTest.java

```diff
@@ -0,0 +1,59 @@
+package bio.terra.service.tags;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.empty;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+
+import bio.terra.common.category.Unit;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.test.context.ActiveProfiles;
+
+@ActiveProfiles({"google", "unittest"})
+@Category(Unit.class)
+public class TagUtilsTest {
+
+  @Test
+  public void testDatasetRequestToDatasetTags() {
+    assertThat("Null tag list is converted to empty list", TagUtils.getDistinctTags(null), empty());
+
+    assertThat("Empty tag list is returned", TagUtils.getDistinctTags(List.of()), empty());
+
+    List<String> nullTagElements = new ArrayList<>();
+    nullTagElements.add(null);
+    assertThat("Null tags are filtered out", TagUtils.getDistinctTags(nullTagElements), empty());
+
+    String tag = "a tag";
+    assertThat(
+        "Duplicate tags are filtered out",
+        TagUtils.getDistinctTags(List.of(tag, tag, tag)),
+        contains(tag));
```

Comment:
I think we should check that the length of `TagUtils.getDistinctTags(List.of(tag, tag, tag))` is 1 here as well.

Reply:
We actually don't need to do that! Here is the documentation for `Matchers.contains`:

```
  /**
   * Creates a matcher for {@link Iterable}s that matches when a single pass over the
   * examined {@link Iterable} yields a series of items, each logically equal to the
   * corresponding item in the specified items.  For a positive match, the examined iterable
   * must be of the same length as the number of specified items.
   * For example:
   * <pre>assertThat(Arrays.asList("foo", "bar"), contains("foo", "bar"))</pre>
   * 
   * @param items
   *     the items that must equal the items provided by an examined {@link Iterable}
   */
```